### PR TITLE
added adjoint check and fixed mkdir for output folder

### LINKED ISSUE #3: Add `checkAdjointSolutionFailure` to opt scripts
## Type of issue
What types of issue is it?
_Select the appropriate type(s) that describe this issue_

- New feature (non-breaking change which adds functionality)

## Description
We should check the adjoint solution in addition to the primal solution check. Need to add the check to the optimization scripts, and run locally to verify.

### PR DESCRIPTION
**Pull request by marcomangano**
## Purpose
I added `checkAdjointFailure()`, so to close #3 
I also took the chance to add an additional check before we build the output directory, so to be sure that the runscript does not freeze if such folder already exists.

## Type of change

- New feature (non-breaking change which adds functionality)
- Maintenance

## Testing
- The script does not freeze if the `output` folder already exists
- now a `'fail'` key is appended to the `funcSens` dictionary, to include in the dict whether the adjoint solver has converged or not.

## Checklist
- [x] I tested the changes locally
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

File: tutorial/airfoilopt/multipoint/airfoil_multiopt.py [hunk 1]
```diff
@@ -36,7 +36,7 @@
 MP.addProcessorSet("cruise", nMembers=nGroup, memberSizes=nProcPerGroup)
 comm, setComm, setFlags, groupFlags, ptID = MP.createCommunicators()
 outputDirectory = "output"
-if comm.rank == 0:
+if comm.rank == 0 and os.path.exists(outputDirectory) is False:
```
Comment (nwu63) [inline tutorial/airfoilopt/multipoint/airfoil_multiopt.py:39]: This doesn't make a lot of sense to me
- The preferred Python syntax is `and not os.path.exists()`
- What happens if the directory exists? Would this overwrite stuff? I think the preferred solution is to exit with a clear error message, so the user either deletes the old directory or specifies a new directory. Blindly overwriting stuff is not a good idea IMO

**Review by nwu63 — CHANGES_REQUESTED** [review]

File: tutorial/airfoilopt/multipoint/airfoil_multiopt.py (hunk 1 above)
Comment (marcomangano) [reply tutorial/airfoilopt/multipoint/airfoil_multiopt.py:39]: No, this just ensures that the directory is created only if not there already, it does not overwrite any file

**Comment by marcomangano** [comment]
I think this can be enough for now actually, we can work on a standard output tree when we advance with the optimization template

File: tutorial/airfoilopt/multipoint/airfoil_multiopt.py [hunk 2]
```diff
@@ -36,8 +36,10 @@
 MP.addProcessorSet("cruise", nMembers=nGroup, memberSizes=nProcPerGroup)
 comm, setComm, setFlags, groupFlags, ptID = MP.createCommunicators()
 outputDirectory = "output"
-if comm.rank == 0 and os.path.exists(outputDirectory) is False:
+if not os.path.exists(outputDirectory):
     os.mkdir(outputDirectory)
+else:
+    raise Warning("The directory already exists! Please delete it or provide a new path")
```
Comment (nwu63) [inline tutorial/airfoilopt/multipoint/airfoil_multiopt.py:42]: Should be an error so the script exits, maybe `OSError`?

Comment (marcomangano) [reply tutorial/airfoilopt/multipoint/airfoil_multiopt.py:42]: Warning exits as well, at least on my machine! Happy to change to `OSError` for consistency though!

Comment (nwu63) [reply tutorial/airfoilopt/multipoint/airfoil_multiopt.py:42]: Ah right, in Python the `Warning` class is derived from the `Exception` class, so by default they will raise an error. However this is less explicit and we should raise a specific error, in this case `OSError` is probably the most appropriate. I'd prefer to leave warnings for things like `DeprecationWarning` etc.

**Review by nwu63 — APPROVED** [review]

**Review by yqliaohk — APPROVED** [review]
Looks good, thanks

**Pull request merged**